src/batch_parser.cc

```diff
@@ -0,0 +1,207 @@
+#include "batch_parser.h"
+
+#include <rocksdb/write_batch.h>
+#include <glog/logging.h>
+
+#include "redis_bitmap.h"
+#include "redis_slot.h"
+#include "util.h"
+
+void WriteBatchExtractor::LogData(const rocksdb::Slice &blob) {
+  log_data_.Decode(blob);
+}
+
+rocksdb::Status WriteBatchExtractor::PutCF(uint32_t column_family_id, const Slice &key,
+                                           const Slice &value) {
+  if (column_family_id == kColumnFamilyIDZSetScore) {
+    return rocksdb::Status::OK();
+  }
+
+  std::string ns, user_key, sub_key;
+  std::vector<std::string> command_args;
+  if (column_family_id == kColumnFamilyIDMetadata) {
+    ExtractNamespaceKey(key, &ns, &user_key, is_slotid_encoded_);
+    if (slot_ >= 0) {
+      if (static_cast<uint16_t>(slot_) != GetSlotNumFromKey(user_key)) return rocksdb::Status::OK();
+    }
+    Metadata metadata(kRedisNone);
+    metadata.Decode(value.ToString());
+    if (metadata.Type() == kRedisString) {
+      command_args = {"SET", user_key, value.ToString().substr(5, value.size() - 5)};
+      aof_strings_[ns].emplace_back(Util::Command2RESP(command_args));
+      if (metadata.expire > 0) {
+        command_args = {"EXPIREAT", user_key, std::to_string(metadata.expire)};
+        aof_strings_[ns].emplace_back(Util::Command2RESP(command_args));
+      }
+    } else if (metadata.expire > 0) {
```

Comment:
should use `else`? whatever key type, you also need to try to set expire time

Reply:
I think it should be `else if` here. No matter what the type of key is, the expiration time of the key needs to be set only when `meatdata.expire > 0` .

Reply:
i mean use `if`, don't need `else`